client/components/domains/use-my-domain/index.jsx

```diff
@@ -101,34 +101,75 @@ function UseMyDomain( {
 		return ! errorMessage;
 	}, [ domainName ] );
 
-	const onNext = useCallback( () => {
+	const setDomainTransferData = useCallback(
+		( isDomainUnlocked, transferEligibleDate ) => {
+			const { LOCKED, UNLOCKED, UNKNOWN } = domainLockStatusType;
+			let lockStatus = UNKNOWN;
+
+			const isTransferrable = null === transferEligibleDate;
```

Comment:
Let's pass the `transferEligibleDate` as a prop to the `DomainTransferOrConnect` that way we can use it in `getOptionInfo` to show a message regarding why the domain isn't transferrable.